src/main/java/io/supertokens/storage/mysql/config/MySQLConfig.java

```diff
@@ -509,10 +509,11 @@ public String getConnectionPoolId() {
         StringBuilder connectionPoolId = new StringBuilder();
         for (Field field : MySQLConfig.class.getDeclaredFields()) {
             if (field.isAnnotationPresent(ConnectionPoolProperty.class)) {
-                connectionPoolId.append("|");
                 try {
-                    if (field.get(this) != null) {
-                        connectionPoolId.append(field.get(this).toString());
+                    String fieldName = field.getName();
+                    String fieldValue = field.get(this) != null ? field.get(this).toString() : null;
+                    if (!fieldName.equals("mysql_password") && fieldValue != null) {
+                        connectionPoolId.append("|" + fieldValue);
```

Comment:
i don't think this is the right thing to do cause the connection id still needs to have the password to uniquely differentiate it from other attempted connections to the same db. So instead, we should change the logging class to not log the password in any string.

Reply:
Fixed.